client/state/data-layer/wpcom-http/utils.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Returns response data from an HTTP request success action if available
+ *
+ * @param {Object} action may contain HTTP response data
+ * @returns {any|null} response data if available
+ */
+export const getData = action => get( action, 'meta.dataLayer.data', null );
+
+/**
+ * Returns error data from an HTTP request failure action if available
+ *
+ * @param {Object} action may contain HTTP response error data
+ * @returns {any|null} error data if available
+ */
+export const getError = action => get( action, 'meta.dataLayer.error', null );
+
+/**
+ * @typedef {Object} ProgressData
```

Comment:
From what I can tell, we should be able to create an external library of types and reuse them from other modules. This would be great in my opinion if we could start doing things like typing our actions as it would give us auto-complete and error messages in our editors/linters which might prevent bugs.

The overhead is on par with a full gradually-typed system I suppose but this way does keep all the code as pure JavaScript.